internal/pkg/node/datastructure.go

```diff
@@ -102,29 +103,52 @@ partition definition, the label must be uniq so its used as the key in the
+	ShouldExistP        *bool  `yaml:"should_exist,omitempty"         json:"should_exist,omitempty"         lopt:"partcreate" comment:"Create partition if it does not exist"`
+	ResizeP             *bool  `yaml:"resize,omitempty"               json:"resize,omitempty"                                 comment:"whether or not the existing partition should be resize"`
 }
 
 /*
 Definition of a filesystem. The device is uniq so its used as key
 */
 type FileSystem struct {
-	id             string   `yaml:"-"                         json:"-"`
-	Format         string   `yaml:"format,omitempty"          json:"format,omitempty"          lopt:"fsformat" comment:"format of the file system"`
-	Path           string   `yaml:"path,omitempty"            json:"path,omitempty"            lopt:"fspath"   comment:"the mount point of the file system"`
-	WipeFileSystem bool     `yaml:"wipe_filesystem,omitempty" json:"wipe_filesystem,omitempty" lopt:"fswipe"   comment:"wipe file system at boot"`
-	Label          string   `yaml:"label,omitempty"           json:"label,omitempty"                           comment:"the label of the filesystem"`
-	Uuid           string   `yaml:"uuid,omitempty"            json:"uuid,omitempty"                            comment:"the uuid of the filesystem"`
-	Options        []string `yaml:"options,omitempty"         json:"options,omitempty"                         comment:"any additional options to be passed to the format-specific mkfs utility"`
-	MountOptions   string   `yaml:"mount_options,omitempty"   json:"mount_options,omitempty"                   comment:"any special options to be passed to the mount command"`
+	id              string   `yaml:"-"                         json:"-"`
+	Format          string   `yaml:"format,omitempty"          json:"format,omitempty"          lopt:"fsformat" comment:"format of the file system"`
+	Path            string   `yaml:"path,omitempty"            json:"path,omitempty"            lopt:"fspath"   comment:"the mount point of the file system"`
+	WipeFileSystemP *bool    `yaml:"wipe_filesystem,omitempty" json:"wipe_filesystem,omitempty" lopt:"fswipe"   comment:"wipe file system at boot"`
+	Label           string   `yaml:"label,omitempty"           json:"label,omitempty"                           comment:"the label of the filesystem"`
+	Uuid            string   `yaml:"uuid,omitempty"            json:"uuid,omitempty"                            comment:"the uuid of the filesystem"`
+	Options         []string `yaml:"options,omitempty"         json:"options,omitempty"                         comment:"any additional options to be passed to the format-specific mkfs utility"`
+	MountOptions    string   `yaml:"mount_options,omitempty"   json:"mount_options,omitempty"                   comment:"any special options to be passed to the mount command"`
 }
 
 type Resource interface{}
+
```

Comment:
Are the  following methods used at all?

Reply:
These methods are primarily for use by overlay templates. Changing the struct field to a pointer would mean that overlay templates have to deal with null values; so the fields are renamed (to a value with a P suffix, meaning "Pointer") and then overlay templates can continue to use the previous names to call these methods which return simple booleans that will always be either true or false.